xls/public/c_api.cc

```diff
@@ -212,6 +212,13 @@ void xls_package_free(struct xls_package* p) {
   delete reinterpret_cast<xls::Package*>(p);
 }
 
+void xls_c_str_free(char* c_str) {
+  if (c_str == nullptr) {
```

Comment:
FYI: No need to compare against nullptr. `free()` is guaranteed to handle the case.